ouroboros-consensus-protocol/src/ouroboros-consensus-protocol/Ouroboros/Consensus/Protocol/Praos/Common.hs

```diff
@@ -59,6 +54,13 @@ data PraosChainSelectView c = PraosChainSelectView
   }
   deriving (Show, Eq, Generic, NoThunks)
 
+-- | We order between chains as follows:
+--
+-- 1. By chain length, with longer chains always preferred.
+-- 2. If the tip of each chain was issued by the same agent, then we prefer
+--    the chain whose tip has the highest ocert issue number.
+-- 3. By a VRF value from the chain tip, with lower values preferred. See
+--    @pTieBreakVRFValue@ for which one is used.
 instance Crypto c => Ord (PraosChainSelectView c) where
```

Comment:
Is it useful to test that it is total, i.e. by using [quickcheck-classes](https://hackage.haskell.org/package/quickcheck-classes)? I'm not sure how useful this is, and how easy it is to generate an arbitrary `PraosChainSelectView`

Reply:
I will definitely do that, that's how I found out that the chain order is not transitive in the first place :sweat_smile:

Reply:
Now implemented :+1: